src/sagemaker/model_card/model_card.py

```diff
@@ -208,6 +208,257 @@ def search_model_associated_model_cards(model_id: str):
+
+        associated_model_cards = search_model_package_associated_model_cards(model_package_response["ModelPackageArn"])
+        if associated_model_cards:
+            raise ValueError(
+                f"The model package has been associated with {associated_model_cards} model cards."
+            )
+
+        kwargs.update(
+            {
+                "model_package_arn": model_package_response["ModelPackageArn"],
+                "model_package_group_name": model_package_response["ModelPackageGroupName"],
+                "model_package_status": model_package_response["ModelPackageStatus"],
+                "model_approval_status": model_package_response["ModelApprovalStatus"],
+                "model_package_version": model_package_response["ModelPackageVersion"],
+            }
+        )
+
+        for key in OPTIONAL_FIELDS_MAP:
+            if key in model_package_response:
+                kwargs.update(
+                    {
+                        OPTIONAL_FIELDS_MAP[key]: model_package_response[key]
+                    }
+                )
+
+        if "UserProfileName" in model_package_response["CreatedBy"]:
+            model_package_creator = ModelPackageCreator(user_profile_name=model_package_response["CreatedBy"]["UserProfileName"])
+            kwargs.update(
+                {
+                    "created_by": model_package_creator
+                }
+            )
+
+        if "InferenceSpecification" in model_package_response:
+            containers_response = model_package_response["InferenceSpecification"]["Containers"]
+            containers = []
+            for container in containers_response:
+                if ("ModelDataUrl" in container and "NearestModelName" in container):
+                    containers.append(Container(container["Image"], container["ModelDataUrl"], container["NearestModelName"]))
+                elif ("ModelDataUrl" in container and "NearestModelName" not in container):
+                    containers.append(Container(container["Image"], container["ModelDataUrl"]))
+                elif ("ModelDataUrl" not in container and "NearestModelName" in container):
+                    containers.append(Container(container["Image"], container["NearestModelName"]))
+                else:
+                    containers.append(Container(container["Image"]))
+
+            kwargs.update(
+                {
+                    "inference_specification": InferenceSpecification(containers)
+                }
+            )
+
+        if "SourceAlgorithmSpecification" in model_package_response:
+            source_algorithms_response = model_package_response["SourceAlgorithmSpecification"]["SourceAlgorithms"]
+            source_algorithms = [
+                SourceAlgorithm(SA["AlgorithmName"], SA["ModelDataUrl"]) if "ModelDataUrl" in SA 
+                else SourceAlgorithm(SA["AlgorithmName"]) 
+                for SA in source_algorithms_response
```

Comment:
in python, local variable in the list comprehension will be in lower case. (SA => sa